include/boost/math/distributions/kolmogorov_smirnov.hpp

```diff
@@ -0,0 +1,494 @@
+// parameterization of the fourth Jacobi Theta function as
+//
+// CDF=θ₄(0|2*x*x*n/pi)
+//
+// The PDF is a hand-coded derivative of that function. Actually, there are two
+// (independent) derivatives, as separate code paths are used for "small x"
+// (2*x*x*n < pi) and "large x", mirroring the separate code paths in the
+// Jacobi Theta implementation to achieve fast convergence. Quantiles are
+// computed using a Newton-Raphson iteration from an initial guess that I
+// arrived at by trial and error.
+//
+// The mean and variance are implemented using simple closed-form expressions.
+// Skewness and kurtosis use slightly more complicated closed-form expressions
+// that involve the zeta function. The mode is calculated at run-time by
+// maximizing the PDF. If you have an analytical solution for the mode, feel
+// free to plop it in.
+//
+// The CDF and PDF could almost certainly be re-implemented and sped up using a
+// polynomial or rational approximation, since the only meaningful argument is
+// x * sqrt(n). But that is left as an exercise for the next maintainer.
+//
+// In the future, the Pelz-Good approximation could be added. I suggest adding
+// a second parameter representing the order, e.g.
+//
+// kolmogorov_smirnov_dist<>(100) // N=100, order=1
+// kolmogorov_smirnov_dist<>(100, 1) // N=100, order=1, i.e. Kolmogorov's formula
+// kolmogorov_smirnov_dist<>(100, 4) // N=100, order=4, i.e. Pelz-Good formula
+//
+// The exact distribution could be added to the API with a special order
+// parameter (e.g. 0 or infinity), or a separate distribution type altogether
+// (e.g. kolmogorov_smirnov_exact_distribution).
+//
+#ifndef BOOST_MATH_DISTRIBUTIONS_KOLMOGOROV_SMIRNOV_HPP
+#define BOOST_MATH_DISTRIBUTIONS_KOLMOGOROV_SMIRNOV_HPP
+
+#include <boost/math/distributions/fwd.hpp>
+#include <boost/math/distributions/complement.hpp>
+#include <boost/math/distributions/detail/common_error_handling.hpp>
+#include <boost/math/special_functions/jacobi_theta.hpp>
+#include <boost/math/tools/tuple.hpp>
+#include <boost/math/tools/roots.hpp> // Newton-Raphson
+#include <boost/math/tools/minima.hpp> // For the mode
+
+namespace boost { namespace math {
+
+namespace detail {
+template <class RealType>
+inline RealType kolmogorov_smirnov_quantile_guess(RealType p) {
+    // Choose a starting point for the Newton-Raphson iteration
+    if (p > 0.9)
+        return RealType(1.8) - 5 * (1 - p);
+    if (p < 0.3)
+        return p + RealType(0.45);
+    return p + RealType(0.3);
+}
+
+// d/dk (theta2(0, 1/(2*k*k/M_PI))/sqrt(2*k*k*M_PI))
+template <class RealType, class Policy>
+RealType kolmogorov_smirnov_pdf_small_x(RealType x, RealType n, const Policy&) {
+    BOOST_MATH_STD_USING
+    RealType value = RealType(0), delta = RealType(0), last_delta = RealType(0);
+    RealType eps = policies::get_epsilon<RealType, Policy>();
+    int i = 0;
+    RealType pi2 = constants::pi_sqr<RealType>();
+    RealType x2n = x*x*n;
```

Comment:
A very minor thing, but I would slap a `const` on these constants to make it easier for the reader and, possibly, the optimizing compiler.

Reply:
Oh, this is already merged, never mind.  :)